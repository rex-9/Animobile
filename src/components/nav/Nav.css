nav {
  font-size: 20px;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 600px;
  background-color: var(--medium);
}

nav > img {
  position: relative;
  right: 150px;
  width: 25px;
  height: 25px;
}

.icons {
  position: relative;
  left: 150px;
  width: 100px;
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.icons img:nth-child(1) {
  width: 36px;
  height: 36px;
}

.icons img:nth-child(2) {
  width: 24px;
  height: 24px;
}