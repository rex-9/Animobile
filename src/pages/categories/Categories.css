h2 {
  text-align: center;
}

.categories {
  display: grid;
  grid-template-columns: 1fr 1fr;
}
