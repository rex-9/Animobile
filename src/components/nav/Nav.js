import './Nav.css';
import back from '../../assets/back.png';
import mic from '../../assets/mic.png';
import setting from '../../assets/setting.png';

const Nav = () => (
  <>
    <nav>
      <img src={back} alt="" />
      <div>Nav</div>
      <div className="icons">
        <img src={mic} alt="" />
        <img src={setting} alt="" />
      </div>
    </nav>
  </>
);

export default Nav;
