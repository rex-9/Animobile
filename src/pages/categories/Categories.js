import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import Card from '../../components/card/Card';
import { fetchCategories } from '../../redux/categories';
import './Categories.css';

const Categories = () => {
  const categories = useSelector((state) => state.categories);
  const dispatch = useDispatch();

  useEffect(() => {
    if (categories.length === 0) {
      dispatch(fetchCategories());
    }
  }, []);

  return (
    <>
      <h2>ðŸ˜‹ Yummy Valley âœ¨</h2>
      <div className="categories">
        {
          categories.map((meal, index) => <Link key={meal.id} to={`/${meal.name}`}><Card index={index} id={meal.id} name={meal.name} thumb={meal.thumb} /></Link>)
        }
      </div>
    </>
  );
};

export default Categories;
