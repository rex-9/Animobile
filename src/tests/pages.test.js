import {
  render, screen,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Categories from '../pages/categories/Categories';
import Nav from '../components/nav/Nav';

describe('Categories page component', () => {
  it('Check if Categories page available', async () => {
    render(
      <Provider store={store}>
        <Categories />
      </Provider>,
    );
    expect(screen.getByText('ðŸ˜‹ Yummy Valley âœ¨')).toMatchSnapshot(); // Categories Page
  });
  it('Check if Meals page available', async () => {
    render(
      <Provider store={store}>
        <Nav />
      </Provider>,
    );
    expect(screen.getByAltText(/mic/i)).toMatchSnapshot(); // Meals Page
  });
  it('Check if Detail page available', async () => {
    render(
      <Provider store={store}>
        <Nav />
      </Provider>,
    );
    expect(screen.getByAltText(/setting/i)).toMatchSnapshot(); // Details Page
  });
});
