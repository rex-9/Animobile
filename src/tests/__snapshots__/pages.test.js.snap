// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories page component Check if Categories page available 1`] = `
<h2>
  ðŸ˜‹ Yummy Valley âœ¨
</h2>
`;

exports[`Categories page component Check if Detail page available 1`] = `
<img
  alt="setting"
  src="setting.png"
/>
`;

exports[`Categories page component Check if Meals page available 1`] = `
<img
  alt="mic"
  src="mic.png"
/>
`;
